src/org/turbanov/execution/cmd/InCmdRunner.java

```diff
@@ -111,39 +108,28 @@ protected RunContentDescriptor doExecute(@NotNull RunProfileState runProfileStat
             newCommandLine = original;
         }
 
-        if (options.isRunInsideTerminal) {
-            boolean ideaTerminalEnabled = isTerminalPluginEnabled();
-            if (!ideaTerminalEnabled) {
-                showWarning("Terminal plugin disabled<br>Run in external cmd instead", environment);
+        if (options.isRunInsideTerminal && isTerminalPluginEnabled()) {
+            try {
+                String[] command = createCommand(false, newCommandLine, null, null);
+                TerminalRunner.runInIdeaTerminal(environment.getProject(), command, classPathPathsString, workingDirectory);
+            } catch (Throwable e) {
+                showWarning("Unable to run in internal IDEA Terminal due to '" + e.getMessage() + "'<br>Run in external cmd instead", environment);
                 runInExternalCmd(classPathPathsString, generalCommandLine, workingDirectory, newCommandLine);
-            } else {
-                try {
-                    String[] command = {"cmd.exe", "/K", newCommandLine};
-                    TerminalRunner.runInIdeaTerminal(environment.getProject(), command, classPathPathsString, workingDirectory);
-                } catch (Throwable e) {
-                    showWarning("Unable to run in internal IDEA Terminal due to '" + e.getMessage() + "'<br>Run in external cmd instead", environment);
-                    runInExternalCmd(classPathPathsString, generalCommandLine, workingDirectory, newCommandLine);
-                }
             }
         } else {
+            if (!isTerminalPluginEnabled()) {
+                showWarning("Terminal plugin disabled<br>Run in external cmd instead", environment);
+            }
             runInExternalCmd(classPathPathsString, generalCommandLine, workingDirectory, newCommandLine);
         }
         return null;
     }
 
-    private static void runInExternalCmd(String classPathPathsString, GeneralCommandLine generalCommandLine, String workingDirectory, String commandLine) throws ProcessNotCreatedException {
-        String[] command;
-        if (SystemInfo.isWindows) {
-            command = new String[]{"cmd.exe", "/C", "\"start cmd.exe /K \"" + commandLine + "\"\""};
-        } else {
-            List<String> parsedLine = ParametersListUtil.parse(commandLine, false, true);
-            ArrayList<String> result = new ArrayList<>(parsedLine);
-            result.add(0, "gnome-terminal");
```

Comment:
can you please bring back `gnome-terminal` support. Use it for in case of Linux

Reply:
Returned.